src/inc/palclr.h

```diff
@@ -484,7 +484,14 @@
 //  and it provides __declspec(selectany) to instruct the linker to merge
 //  duplicate external const static data copies into one.
 //  
+#if defined(__clang__)
+#define SELECTANY extern
```

Comment:
I get the following error without this change:

C:/michelm/coreclr/src/jit/_typeinfo.h:62:1: error: 'selectany' can only be applied to data items with external linkage [clang-diagnostic-error]
SELECTANY const ti_types g_jit_types_map[] = {
^
C:/michelm/coreclr/src/jit/../inc/palclr.h:487:37: note: expanded from macro 'SELECTANY'
#define SELECTANY extern __declspec(selectany). I'll try with PLATFORM_UNIX defined.